security_group.tf

```diff
@@ -0,0 +1,77 @@
+# Security group for IPA servers
+resource "aws_security_group" "ipa_servers" {
+  count = var.is_master ? 1 : 0
+
+  vpc_id = data.aws_subnet.the_subnet.vpc_id
+
+  description = "Security group for IPA servers"
+  tags        = var.tags
+}
+
+# TCP ingress rules for IPA
+resource "aws_security_group_rule" "ipa_tcp_ingress_trusted" {
+  count = var.is_master ? length(local.ipa_tcp_ports) : 0
+
+  security_group_id = aws_security_group.ipa_servers[0].id
+  type              = "ingress"
+  protocol          = "tcp"
+  cidr_blocks       = var.trusted_cidr_blocks
+  from_port         = local.ipa_tcp_ports[count.index]
+  to_port           = local.ipa_tcp_ports[count.index]
+}
+resource "aws_security_group_rule" "ipa_tcp_ingress_self" {
+  count = var.is_master ? length(local.ipa_tcp_ports) : 0
+
+  security_group_id = aws_security_group.ipa_servers[0].id
+  type              = "ingress"
+  protocol          = "tcp"
+  self              = true
+  from_port         = local.ipa_tcp_ports[count.index]
+  to_port           = local.ipa_tcp_ports[count.index]
+}
+
+# TCP egress rules for IPA
+resource "aws_security_group_rule" "ipa_tcp_egress" {
```

Comment:
Should this rule be renamed `ipa_tcp_egress_self` for consistency with `ipa_tcp_ingress_self`?  Same thing below for `ipa_udp_egress_self`.

Reply:
Corrected in 8db8976.